CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/resourcetypes/CVKImage.java

```diff
@@ -455,28 +600,353 @@ public static CVKImage Create(  CVKDevice cvkDevice,
+            byte[] rgb = new byte[(int)fileSize];
+            int i = 0;
+            int r = 0;
+
+            ByteBuffer row = imageMemory.getByteBuffer((int)sourceImageSize);
+
+            for (int y = 0; y < fileHeight; y++)
+            {                   
+                for (int x = 0; x < fileWidth; x++)
+                {
+                    // Skip the padded bytes
+                    if (x >= width) {
+                        r += 4;
+                        continue;
+                    }
+                    // TODO: tests to see if we need to swizzle the rgb channels
+                    Byte red = row.get(r++);        // red
+                    Byte green = row.get(r++);      // green
+                    Byte blue = row.get(r++);       // blue
+                    Byte alpha = row.get(r++);      // alpha
+
+                    // If we need to swizzle colours manually (e.g. Blit is not supported which
+                    // does automatic conversion for us
+                    if (needsSwizzle) {
+                        rgb[i++] = blue;
+                        rgb[i++] = green;
+                        rgb[i++] = red;
+                    }
+                    else {
+                        rgb[i++] = red;
+                        rgb[i++] = green;
+                        rgb[i++] = blue;
+                    }
+                }
+            }
+
+            DataBuffer buffer = new DataBufferByte(rgb, rgb.length);
+
+            // 3 bytes per pixel: red, green, blue
+            WritableRaster raster = Raster.createInterleavedRaster(buffer, 
+                    width, height,          // width, height of image
+                    3 * width,              // number of bytes per row (stride)
+                    3,                      // number of channels (rgb)
+                    new int[] {0, 1, 2},    // colour format order (i.e. byte 0 is first, byte 1 is second ...
+                                            // can change order if you want a different format rgb -> bgr
+                    (Point)null);           // TODO Look up this
+            ColorModel cm = new ComponentColorModel(ColorModel.getRGBdefault().getColorSpace(), false, true, Transparency.OPAQUE, DataBuffer.TYPE_BYTE); 
+            BufferedImage image = new BufferedImage(cm, raster, true, null);
+
+            // Save to file
+            ImageIO.write(image, "png", file);
+        }
+        catch (Exception e) {
+            System.err.println("Error: " + e.getMessage());
+            cvkDevice.GetLogger().warning(String.format("Save file failed: %s", file.getName()));
+            return CVK_ERROR_SAVE_TO_FILE_FAILED;
+        }
+
+        return ret;
+    }
+    
+    
+    /**
+     * Cleanup the mapped memory from StartMapMemory
+     * 
+     * @param imageSource 
+     */
+    public void EndMapMemory(CVKImage imageSource) {    
+        vkUnmapMemory(cvkDevice.GetDevice(), imageSource.GetMemoryImageHandle());
```

Comment:
Does the pointer need to be freed?

Reply:
Don't think so as it is allocated on the stack (freeing it causing mem exception)
I have added code to destroy the destImage in SaveToFile though

Reply:
Oh if it is allocated on the 'stack' it should not be returned from the function!  Have a look at CVKBuffer.StartWrite

Reply:
Rewrote the functions a little bit, but all fixed now